src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        final int arrayLength = 10;
+        Figure[] figures = new Figure[arrayLength];
+        FigureSupplier figureSupplier = new FigureSupplier();
+        for (int i = 0; i < arrayLength; i++) {
+            if (i < arrayLength / 2) {
+                figures[i] = figureSupplier.getRandomFigure();
+            } else {
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
+            System.out.println(figures[i].draw());
+        }
```

Comment:
Instead of using System.out.println() directly in the loop, consider collecting the drawing information in a String or StringBuilder and outputting it once after the loop. This approach is usually more efficient and can be easier to manage in larger applications.